be/src/formats/parquet/parquet_file_writer.h

```diff
@@ -93,36 +98,41 @@ class ParquetFileWriter final : public FileWriter {
     const std::vector<std::string> _column_names;
     const std::vector<TypeDescriptor> _type_descs;
     std::vector<std::unique_ptr<ColumnEvaluator>> _column_evaluators;
+    TCompressionType::type _compression_type;
     std::shared_ptr<ParquetWriterOptions> _writer_options;
     std::function<StatusOr<ColumnPtr>(Chunk*, size_t)> _eval_func;
 
     std::shared_ptr<::parquet::ParquetFileWriter> _writer;
     std::shared_ptr<parquet::ChunkWriter> _rowgroup_writer;
     const std::function<void()> _rollback_action;
     PriorityThreadPool* _executors;
+    RuntimeState* _runtime_state;
 };
 
 class ParquetFileWriterFactory : public FileWriterFactory {
 public:
-    ParquetFileWriterFactory(std::shared_ptr<FileSystem> fs, const std::map<std::string, std::string>& options,
+    ParquetFileWriterFactory(std::shared_ptr<FileSystem> fs, TCompressionType::type compression_type,
+                             const std::map<std::string, std::string>& options,
                              const std::vector<std::string>& column_names,
                              std::vector<std::unique_ptr<ColumnEvaluator>>&& column_evaluators,
-                             std::optional<std::vector<formats::FileColumnId>> field_ids = std::nullopt,
-                             PriorityThreadPool* executors = nullptr);
+                             std::optional<std::vector<formats::FileColumnId>> field_ids, PriorityThreadPool* executors,
+                             RuntimeState* runtime_state);
 
     Status init() override;
 
     StatusOr<std::shared_ptr<FileWriter>> create(const std::string& path) override;
 
 private:
     std::shared_ptr<FileSystem> _fs;
+    TCompressionType::type _compression_type;
     std::optional<std::vector<formats::FileColumnId>> _field_ids;
     std::map<std::string, std::string> _options;
     std::shared_ptr<ParquetWriterOptions> _parsed_options;
 
     std::vector<std::string> _column_names;
     std::vector<std::unique_ptr<ColumnEvaluator>> _column_evaluators;
     PriorityThreadPool* _executors;
+    RuntimeState* _runtime_state;
```

Comment:
all pointer members should set to nullptr

Reply:
done